### PR TITLE
CHAD-9222 IAS Zone/garage door interaction

### PR DESCRIPTION
**Pull request by greens**
Stop generating contact events from IAS zone messages if the device is installed on a garage door

### CODE REVIEW

**Comment by varzac** [comment]
I would have expected changes to be necessary in https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-contact/src/smartsense-multi/init.lua but looking at it now, I don't see any handling of garageSensor in there.  Do you think that deserves a separate ticket?

File: drivers/SmartThings/zigbee-contact/src/multi-sensor/init.lua [hunk 1]
```diff
@@ -49,6 +51,18 @@ local function multi_sensor_report_handler(driver, device, zb_rx)
...
 end
 
+local function zone_status_change_handler(driver, device, zb_rx)
+  if not device.preferences["garageSensor"] then
+    contactSensor_defaults.ias_zone_status_change_handler(driver, device, zb_rx)
+  end
+end
+
+local function zone_status_handler(driver, device, zone_status, zb_rx)
+  if not device.preferences["garageSensor"] then
```
[the remark this answers is not in the document]
Comment (varzac) [reply drivers/SmartThings/zigbee-contact/src/multi-sensor/init.lua:61]: I can try to install the driver from this PR today and do some testing today if I can find some spare batteries for my extra sensors.

**Review by varzac — APPROVED** [review]
I have tested that this works as expected.

**Comment by greens** [comment]
@varzac you're right about the separate smartsense-multi driver. The garage door code is in `multi_utils` which is why you didn't see it

**Review by varzac — APPROVED** [review]

**Comment by timpatten** [comment]
For some reason the Garage Door feature no longer works.  If you have this enabled, no matter what direction I move the sensor, it will never say Closed.  I know it was working prior to this update but i'm not sure how reliable it was.  New issue or bug with this update?

**Comment by varzac** [comment]
@timpatten we are not aware of any issues and have reports of this fixing the previous issue of the contact state flip flopping when in the garage door mode.  If you reach out to Nayely on the SmartThings community and provide hub logs along with times of when it didn't trigger when you think it should have she can create an internal ticket for us to investigate.

**Pull request merged**